deploy/firesim

```diff
@@ -114,15 +115,15 @@ def buildafi(globalbuildconf):
     def instance_liveness():
         """ confirm that all instances are running first. """
         rootLogger.info("""[{}] Checking if host instance is up...""".format(env.host_string))
-        with StreamLogger('stdout'), StreamLogger('stderr'):
-            run("uname -a")
+        run("uname -a")
 
     auto_create_bucket(globalbuildconf.s3_bucketname)
 
     def terminate_instances_handler(sig, frame):
         """ Handler that prompts to terminate build instances if you press ctrl-c. """
         rootLogger.info("You pressed ctrl-c, so builds have been killed.")
-        userconfirm = raw_input("Do you also want to terminate your build instances? Type 'yes' to do so.\n")
+        with InfoStreamLogger('stdout'):
+            userconfirm = raw_input("Do you also want to terminate your build instances? Type 'yes' to do so.\n")
```

Comment:
I thought I confirmed that wrapping this in `InfoStreamLogger('stdout')` would cause the prompt to go to the console but it definitely didn't when I ran this manually.  Need to look into this a bit more.

Reply:
`raw_input` doesn't print a newline at the end of it's message and my tests were all printing whole lines ending in `\n`.  I tried using `unbuffered=True` in my `InfoStreamLogger` but it looks like `StreamLogger` has a 'feature' where it buffers output by lines even if you have asked for `unbuffered` output.  I can understand why it does that since logging is a linewise operation (to properly format the log with whatever prefix you've configured).  However, it seems almost like a bug because how are you supposed to log things that don't contain newlines???

To work around this, I've determined that explicitly adding a newline at the end of `raw_input()` prompt messages will make them work again (i.e. be displayed to the console when prompting for input).

Finally, it seems like there can be contents left in the `StreamLogger` buffer on context exit that never gets logged (if, for example it didn't end in a newline).  That, seems like a bug to me.

Reply:
This thread is really confusing because I'm pointing at a call to `raw_input` that does end it's prompt with `\n`.  I intended to comment on lines 86 &87  where the prompt doesn't end in `\n` and where I was noticing the confusing behavior.

@sagark  and @davidbiancolin , I think that the cleanest way to fix this would be to create a `firesim_input()` helper function that wraps `raw_input()` api,  explicitly logs the message string at info level and then calls `raw_input()` with the remaining arguments but an empty prompt.  Depending on people to remember to end their `raw_input()` prompt strings in newline seems way too brittle.  This would also refactor all calls of `raw_input()` in firesim manager into one spot that will be easier to update to work in python3 (because raw_input goes away).